src/frontend/js/components/DashboardAddressesManagement/DashboardEditAddress.spec.tsx

```diff
@@ -0,0 +1,175 @@
+import { hydrate, QueryClientProvider } from 'react-query';
+import fetchMock from 'fetch-mock';
+import { act } from '@testing-library/react-hooks';
+import { findByText, fireEvent, render, screen } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import * as mockFactories from 'utils/test/factories';
+import createQueryClient from 'utils/react-query/createQueryClient';
+import { SessionProvider } from 'data/SessionProvider';
+import { DashboardTest } from 'components/Dashboard/DashboardTest';
+import { DashboardPaths } from 'utils/routers/dashboard';
+import { expectFetchCall } from 'utils/test/expectFetchCall';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+describe('<DashboardEditAddress/>', () => {
+  const createQueryClientWithUser = (isAuthenticated: Boolean) => {
+    const user = isAuthenticated ? mockFactories.UserFactory.generate() : null;
+    const { clientState } = mockFactories.PersistedClientFactory({
+      queries: [mockFactories.QueryStateFactory('user', { data: user })],
+    });
+    const client = createQueryClient();
+    hydrate(client, clientState);
+
+    return client;
+  };
+
+  beforeEach(() => {
+    fetchMock.get('https://joanie.endpoint/api/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', []);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetchMock.restore();
+    sessionStorage.clear();
```

Comment:
```suggestion
```

Reply:
we use `useSessionQuery` which uses session it self, that's why I kept it here in order to reset react-query internal store to make the tests independent from each other. What do you think ?